java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -241,94 +377,110 @@ public static void jdbcToArrowVectors(ResultSet rs, VectorSchemaRoot root, Calen
-                    rs.getLong(i), !rs.wasNull(), rowCount);
-            break;
-          case Types.NUMERIC:
-          case Types.DECIMAL:
-            updateVector((DecimalVector) root.getVector(columnName),
-                    rs.getBigDecimal(i), !rs.wasNull(), rowCount);
-            break;
-          case Types.REAL:
-          case Types.FLOAT:
-            updateVector((Float4Vector) root.getVector(columnName),
-                    rs.getFloat(i), !rs.wasNull(), rowCount);
-            break;
-          case Types.DOUBLE:
-            updateVector((Float8Vector) root.getVector(columnName),
-                    rs.getDouble(i), !rs.wasNull(), rowCount);
-            break;
-          case Types.CHAR:
-          case Types.NCHAR:
-          case Types.VARCHAR:
-          case Types.NVARCHAR:
-          case Types.LONGVARCHAR:
-          case Types.LONGNVARCHAR:
-            updateVector((VarCharVector) root.getVector(columnName),
-                    rs.getString(i), !rs.wasNull(), rowCount);
-            break;
-          case Types.DATE:
-            updateVector((DateMilliVector) root.getVector(columnName),
-                    rs.getDate(i, calendar), !rs.wasNull(), rowCount);
-            break;
-          case Types.TIME:
-            updateVector((TimeMilliVector) root.getVector(columnName),
-                    rs.getTime(i, calendar), !rs.wasNull(), rowCount);
-            break;
-          case Types.TIMESTAMP:
-            // TODO: Need to handle precision such as milli, micro, nano
-            updateVector((TimeStampVector) root.getVector(columnName),
-                    rs.getTimestamp(i, calendar), !rs.wasNull(), rowCount);
-            break;
-          case Types.BINARY:
-          case Types.VARBINARY:
-          case Types.LONGVARBINARY:
-            updateVector((VarBinaryVector) root.getVector(columnName),
-                    rs.getBinaryStream(i), !rs.wasNull(), rowCount);
-            break;
-          case Types.ARRAY:
-            // TODO Need to handle this type
-            // fields.add(new Field("list", FieldType.nullable(new ArrowType.List()), null));
-            break;
-          case Types.CLOB:
-            updateVector((VarCharVector) root.getVector(columnName),
-                    rs.getClob(i), !rs.wasNull(), rowCount);
-            break;
-          case Types.BLOB:
-            updateVector((VarBinaryVector) root.getVector(columnName),
-                    rs.getBlob(i), !rs.wasNull(), rowCount);
-            break;
-
-          default:
-            // no-op, shouldn't get here
-            break;
-        }
+        jdbcToFieldVector(
+            rs,
+            i,
+            rs.getMetaData().getColumnType(i),
+            rowCount,
+            root.getVector(rsmd.getColumnName(i)),
+            config);
       }
       rowCount++;
     }
     root.setRowCount(rowCount);
   }
 
+  private static void jdbcToFieldVector(
+      ResultSet rs,
+      int i,
```

Comment:
a name indicating this is the column index might be a little clearer.

Reply:
Fair point.